feat: Use centralized approach for per-repository file caches (#10618)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

This configuration will be applied after all other environment variables so that it can be used to override defaults.

## dockerCache

> When set to 'volume', Renovate creates and mounts a temporary Docker volume that is used as a cache for package manager files.
> 
> When it's 'mount', temporary directory is created and mounted instead of volume.
> 
> Cache will be removed after repository processing. To disable per-repo caching for Docker containers, use 'none' (default).

Reviewer: ```suggestion
When set to 'volume', Renovate creates and mounts a temporary Docker volume that is used as a cache for package manager files.
When set to 'mount', a temporary directory is created and mounted instead.

The cache will be removed after repository processing.
To disable per-repo caching for Docker containers, use 'none' (default).
```

This configuration will be applied after all other environment variables so that it can be used to override defaults.

## dockerCacheVolume

> Create and mount volume for manager cache files. Will be removed after repository processing.

Reviewer: ```suggestion
Creates and mounts a temporary Docker volume that is used as a cache for package manager files.
This volume will be removed after repository processing.
```

How about something like this?

Please check if my description is technically correct. 😉

Author: Thanks a lot!

> 

Author: @HonkingGoose Can you please tell... well, how to make it sound well :)

This configuration will be applied after all other environment variables so that it can be used to override defaults.

## dockerCache

When set to 'volume', Renovate creates and mounts a temporary Docker volume that is used as a cache for package manager files.

> When set to 'mount', a temporary directory is created and mounted instead.

Reviewer 2: Why were you thinking we need both? I don't think we've described enough why someone would want to use this OR how they would choose which. I'm not sure we need both, but open to it

Author: Once I figured out how to mount cache volumes properly, it became clear to me that folder-based mounting implementation is a low-hanging fuit. I agree, we need to point out more clearly it's a non-mainstream option. Basically, it may be used only when volumes are problematic for some reason.
